Gems/PhysX/Code/Editor/Source/ComponentModes/Joints/JointsComponentMode.cpp

```diff
@@ -0,0 +1,598 @@
+        static const char* TwistLimitsTitle = "Switch to Twist Limits Mode";
+        static const char* TwistLimitsToolTip = "Twist Limits Mode - Change the limits of the joint.";
+
+        const AZ::Crc32 SwitchToSwingLimitsSubMode = AZ_CRC_CE("com.o3de.action.physx.joints.switchtoswinglimits");
+        static const char* SwingLimitsTitle = "Switch to Swing Limits Mode";
+        static const char* SwingLimitsToolTip = "Swing Limits Mode - Change the limits of the joint.";
+
+        const AZ::Crc32 SwitchToSnapPositionSubMode = AZ_CRC_CE("com.o3de.action.physx.joints.switchtosnapposition");
+        static const char* SnapPositionTitle = "Switch to Snap Position Mode";
+        static const char* SnapPositionToolTip = "Snap Position Mode - Snap the position of the joint to another Entity.";
+
+        const AZ::Crc32 SwitchToSnapRotationSubMode = AZ_CRC_CE("com.o3de.action.physx.joints.switchtosnaprotation");
+        static const char* SnapRotationTitle = "Switch to Snap Rotation Mode";
+        static const char* SnapRotationToolTip = "Snap Rotation Mode - Snap the rotation of the joint toward another Entity.";
+
+        const AZ::Crc32 ResetSubMode = AZ_CRC_CE("com.o3de.action.physx.joints.resetsubmode");
+        static const char* ResetTitle = "Reset Current Mode";
+        static const char* ResetToolTip = "Reset All changes made during this edit.";
+
+    } // namespace ActionData
+
+    namespace Internal
+    {
+        static AzToolsFramework::ViewportUi::ButtonId RegisterClusterButton(
+            AzToolsFramework::ViewportUi::ClusterId clusterId, const char* iconName, const char* tooltip)
+        {
+            AzToolsFramework::ViewportUi::ButtonId buttonId;
+            AzToolsFramework::ViewportUi::ViewportUiRequestBus::EventResult(
+                buttonId, AzToolsFramework::ViewportUi::DefaultViewportId,
+                &AzToolsFramework::ViewportUi::ViewportUiRequestBus::Events::CreateClusterButton, clusterId,
+                AZStd::string::format(":/stylesheet/img/UI20/toolbar/%s.svg", iconName));
+
+            AzToolsFramework::ViewportUi::ViewportUiRequestBus::Event(
+                AzToolsFramework::ViewportUi::DefaultViewportId,
+                &AzToolsFramework::ViewportUi::ViewportUiRequestBus::Events::SetClusterButtonTooltip, clusterId, buttonId, tooltip);
+
+            return buttonId;
+        }
+
+        void RefreshUI()
+        {
+            /// The reason this is in a free function is because ColliderComponentMode
+            /// privately inherits from ToolsApplicationNotificationBus. Trying to invoke
+            /// the bus inside the class scope causes the compiler to complain it's not accessible
+            /// to due private inheritance.
+            /// Using the global namespace operator :: should have fixed that, except there
+            /// is a bug in the Microsoft compiler meaning it doesn't work. So this is a work around.
```

Comment:
uber nit: This can just be `//` comment style (thanks for adding it though!) - if it helps we could make the notification bus inherit protected instead? We just don't want it to be public

Reply:
oh, this is just a move from a different location. i missed updating to use the ```//```

Reply:
Nice one thanks! :+1: